### PR TITLE
New EnumerateFunction*.getMarginal()

### PR DESCRIPTION
**Pull request by mbaudin47**
- New LinearEnumerateFunction.getMarginal()
- New HyperbolicAnisotropicEnumerateFunction.getMarginal() 
- New NormInfEnumerateFunction.getMarginal()

### CODE REVIEW

File: lib/src/Base/Func/HyperbolicAnisotropicEnumerateFunction.cxx [hunk 1]
```diff
@@ -275,6 +276,18 @@ void HyperbolicAnisotropicEnumerateFunction::setUpperBound(const Indices & upper
   initialize();
 }
 
+/* The marginal enumerate function */
+EnumerateFunction HyperbolicAnisotropicEnumerateFunction::getMarginal(const Indices & indices) const
+{
+  const UnsignedInteger inputDimension = getDimension();
+  indices.check(inputDimension);
```
Comment (jschueller) [inline lib/src/Base/Func/HyperbolicAnisotropicEnumerateFunction.cxx:283]: check just returns a boolean, you have to throw if false

File: lib/src/Base/Func/LinearEnumerateFunction.cxx [hunk 2]
```diff
@@ -174,6 +175,16 @@ void LinearEnumerateFunction::setUpperBound(const Indices & /*upperBound*/)
   throw NotYetImplementedException(HERE) << " in LinearEnumerateFunction::setUpperBound";
 }
 
+/* The marginal enumerate function */
+EnumerateFunction LinearEnumerateFunction::getMarginal(const Indices & indices) const
+{
+  const UnsignedInteger inputDimension = getDimension();
+  indices.check(inputDimension);
```
Comment (jschueller) [inline lib/src/Base/Func/LinearEnumerateFunction.cxx:182]: same

File: lib/src/Base/Func/NormInfEnumerateFunction.cxx [hunk 3]
```diff
@@ -143,6 +144,16 @@ UnsignedInteger NormInfEnumerateFunction::getMaximumDegreeStrataIndex(const Unsi
   return maximumDegree / getDimension();
 }
 
+/* The marginal enumerate function */
+EnumerateFunction NormInfEnumerateFunction::getMarginal(const Indices & indices) const
+{
+  const UnsignedInteger inputDimension = getDimension();
+  indices.check(inputDimension);
```
Comment (jschueller) [inline lib/src/Base/Func/NormInfEnumerateFunction.cxx:151]: and here

**Review by jschueller — APPROVED** [review]
great!

File: lib/src/Base/Func/HyperbolicAnisotropicEnumerateFunction.cxx (hunk 1 above)
Comment (mbaudin47) [reply lib/src/Base/Func/HyperbolicAnisotropicEnumerateFunction.cxx]: Very true: fixed at a3e4351f503ea9976732960cfb0f4917c6b05d94.

File: lib/src/Base/Func/NormInfEnumerateFunction.cxx (hunk 3 above)
Comment (mbaudin47) [reply lib/src/Base/Func/NormInfEnumerateFunction.cxx]: Very true: fixed at a3e4351f503ea9976732960cfb0f4917c6b05d94.

**Comment by mbaudin47** [comment]
@regislebrun : Can you please review this PR?

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**